libs/langgraph/langgraph/utils.py

```diff
@@ -183,3 +189,20 @@ def coerce_to_runnable(thing: RunnableLike, *, name: str, trace: bool) -> Runnab
             f"Expected a Runnable, callable or dict."
             f"Instead got an unsupported type: {type(thing)}"
         )
+
+
+def is_optional_type(type_: Any) -> bool:
+    """Check if a type is Optional."""
+
+    if hasattr(type_, "__origin__") and hasattr(type_, "__args__"):
+        origin = get_origin(type_)
+        if origin is Optional:
+            return True
+        if origin is Union:
+            return any(
+                arg is type(None) or is_optional_type(arg) for arg in type_.__args__
```

Comment:
why `arg is type(None)` instead of `arg is None`

Reply:
ya i think i typoed that when i was inpsecting and seeing "NoneType"

Reply:
[edit] It actually *would* have to be nonetype rather than None check directly